src/shell/dotnet/Shell/App.xaml.cs

```diff
@@ -121,11 +120,11 @@ private async Task StartAsync(StartupEventArgs e)
         var diagnostics = new DiagnosticInfo
         {
             StartupTime = DateTime.Now,
-            ShellVersion = Assembly.GetExecutingAssembly().FullName
+            ShellVersion = Assembly.GetExecutingAssembly().FullName ?? throw new ArgumentNullException(nameof(Assembly.GetExecutingAssembly))
```

Comment:
`ArgumentNullException` should be thrown when a required method argument is null.
What do you want to achieve here?